File: database/notification.go
package database

type NotificationType int

const (
	MessageChatType NotificationType = iota
	InfoType
)

type Notification struct {
	ID        int64  `json:"id"`
	UserID int64 `json:"user_id"`
	Title string `json:"title"`
	Message string `json:"message"`
	NotificationType NotificationType `json:"notification_type"`
	CreatedAt string `json:"created_at"`
}

File: NkataAndroid/app/src/main/java/com/ifeanyi/nkataandroid/logic/database/room/model/Notification.kt
package com.ifeanyi.nkataandroid.logic.database.room.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "notification")
data class Notification(
    @PrimaryKey(autoGenerate = true)
    val id: Int? = null,
    val cloudId: Int? = null,
    val userId: String,
    val username: String,
    val tile:  String,
    val message: String,
    val seen: Boolean,
    val createdAt: String,
    val modifiedAt: String
) {
}